app/services/tagging/tagging_update_form.rb

```diff
@@ -40,6 +41,24 @@ def related_item_errors
       @related_item_errors ||= {}
     end
 
+    def title_for_related_link(base_path)
+      link = links.ordered_related_items.find { |related_item| related_item.fetch("base_path") == base_path }
+
+      if link.nil?
+        # links is populated from the existing links. When we submit a form and
+        # it fails, we may have additional related links we want to know the
+        # titles of.
+        # In practice this shouldn't actually be visible most of the time,
+        # because with javascript enabled, the user can't enter invalid data,
+        # and without it we don't display the titles. It is still possible to
+        # see errors due to tagging conflicts though.
+        # TODO: attempt to fetch missing links on demand.
```

Comment:
I don't know if this is relevant, but I'm about to start adding the feature to manually edit taxon URLs (https://trello.com/c/5cBp6Ryt/303-allow-taxon-urls-to-be-edited). 

This will require detection of URL collision (and appropriate error handling), so I could look into this on that card? (Although it also means there could be impending merge conflicts?)